lib/model/CDetectionRule.cc

```diff
@@ -54,6 +65,31 @@ bool CDetectionRule::apply(ERuleAction action,
     return true;
 }
 
+void CDetectionRule::executeCallback(CAnomalyDetectorModel& model, core_t::TTime time) const {
+
+    if (m_Callback) {
```

Comment:
We tend to avoid implicit conversions
```suggestion
    if (m_Callback != std::nullptr) {
```

Reply:
👍🏻